metalware-new/src/templater.rb

```diff
@@ -199,70 +211,93 @@ def method_missing(group_symbol)
     end
   end
 
-  MagicNamespace = Struct.new(:index, :nodename, :firstboot) do
-    def initialize(index: 0, nodename: nil, firstboot: nil)
-      super(index, nodename, firstboot)
-    end
-
-    def genders
-      # XXX Do we want to make genders available as a `Hashie::Mash` too?
-      # Depends if we want to be able to iterate through genders or just get
-      # list of nodes in a specified gender
-      GenderGroupProxy
+  class MagicNamespace
+    def initialize(*a)
+      @missing_tags = []
+      @struct = MagicNamespaceStruct.new(*a)
     end
 
-    def hunter
-      if File.exists? Constants::HUNTER_PATH
-        Hashie::Mash.load(Constants::HUNTER_PATH)
+    def method_missing(s, *a, &b)
+      if s == :[]
+        value = (a.length == 1 ? @struct[a[0]] : a.map { |v| @struct[v] })
```

Comment:
It's not clear to me what the bug you're trying to fix here is, can you expand on what this is/when this is caused?

Also, why do we need to special case a one element array here? A one element array can be `map`ped the same as one with two or zero or any other number of elements, unless I'm missing something here.